src/logging.rs

```diff
@@ -337,6 +353,13 @@ impl<C: fmt::Display, D: fmt::Display> Client<C, D> {
         }
     }
 
+    pub fn with_tls(self, tls: TlsStatus) -> Self {
```

Comment:
Ditto.

(It seems wrong that we have to implement this twice.)

Reply:
> (It seems wrong that we have to implement this twice.)

Yeah, I think it might be worth factoring out the common code between the `Client` and `Server` contexts somehow, but I feel like that runs the risk of making it more complex overall?

Reply:
> Yeah, I think it might be worth factoring out the common code between the Client and Server contexts somehow, but I feel like that runs the risk of making it more complex overall?

I think there are probably relatively simple ways of doing it, but hard to say without trying it. It doesn't need to be done in this PR, for sure.